numpy/core/src/multiarray/alloc.c

```diff
@@ -35,6 +35,12 @@ typedef struct {
 static cache_bucket datacache[NBUCKETS];
 static cache_bucket dimcache[NBUCKETS_DIM];
 
+#if defined(NPY_PY3K)
+#define NPY_CHECK_GIL_HELD() PyGILState_Check()
+#else
+#define NPY_CHECK_GIL_HELD() 1
```

Comment:
one could implement a check for py2 based on the code in the py3 function, but it doesn't really seem worth it at this point